lib/gitsh/git_repository.rb

```diff
@@ -8,7 +8,8 @@ def initialize(env)
     end
 
     def initialized?
-      git_dir && File.exist?(git_dir)
+      dir = git_dir
+      dir && File.exist?(dir)
```

Comment:
I'm confused in all these suggestions -- not sure what I'm missing -- but what is the `TypeError` that would be raised? Can `git_output` produce `nil`?